Server/src/services/fileUploadService.ts

```diff
@@ -1,118 +1,380 @@
+    "material": {
+      "$ref": "/materialSchema"
+    },
+    "category": { "type": "string" },
+    "subcategory": { "type": "string" },
+    "data": {
+      "$ref": "/dataSchema"
+    }
+
+  },
+  "required": ["reference", "datasetName", "material", "data"]
+
+};
+
+var referenceSchema = {
+  "id": "/referenceSchema",
+  "type": "object",
+  "properties": {
+    "type": { "type": "string" },
+    "publisher": { "type": "string" },
+    "authors": {
+      "$ref": "/authorsSchema"
+    },
+    "title": { "type": "string" },
+    "volume": { "type": "integer" },
+    "pages": { "type": "integer" },
+    "year": { "type": "integer" }
+  },
+  "required": ["authors"]
+};
+
+var materialSchema = {
+  "id": "/materialSchema",
+  "type": "array",
+  "items": {
+    "$ref": "/m2Schema"
+  }
+};
+
+var m2schema = {
+  "id": "/m2Schema",
+  "type": "object",
+  "properties": {
+    "composition": { "type": "string" },
+    "details": { "type": "string" }
+  },
+  "required": ["composition", "details"]
+};
+
+var dataSchema = {
+  "id": "/dataSchema",
+  "type": "object",
+  "properties": {
+    "variables": {
+      "$ref": "/variableSchema"
+    },
+    "contents": {
+      "$ref": "/contentsSchema"
+    },
+    "comments": { "type": "string" }
+  }
+};
+
+var variableSchema = {
+  "id": "/variableSchema",
+  "type": "array",
+  "properties": {
+    "$ref": "/v2Schema"
+  }
+};
+
+var v2Schema = {
+  "id": "/v2Schema",
+  "type": "object",
+  "properties": {
+    "name": { "type": "string" },
+    "repr": { "type": "string" },
+    "units": { "type": "string" }
+  },
+  "required": ["name", "repr", "units"]
+};
+
+var contentsSchema = {
+  "id": "/contentsSchema",
+  "type": "array",
+  "properties": {
+    "$ref": "/c2Schema"
+  }
+};
+
+var c2Schema = {
+  "id": "/c2Schema",
+  "type": "object",
+  "properties": {
+    "points": {
+      "type": "array",
+      "items": { "type": "number" },
+    },
+    "comments": { "type": "string" }
+  },
+  "required": ["points"]
+};
+
+/**
+ * The methods in this class are only responsible for processing uploaded files. Input will be parsed 
+ * then stored into its appropriate table in the database. 
+ */
+
+const processUpload = async (filePathOfJson) => {
+  let response = jsonUpload(filePathOfJson);
+  return response;
+}
+
+const jsonUpload = async (filePathOfJson) => {
+
+  let category = '';
+  let subcategory = '';
+  let dataSetName = '';
+  let dataType = '';
+  let dataSetComments = '';
+  let individualDataSetComments = [];
+  let material = [];
+  let referenceType = '';
+  let referencePublisher: string = '';
+  let referenceTitle = 'hello';
+  let referenceAuthors = [];
+  let referenceYear;
+  let referencePages;
+  let referenceVolume;
+  let referenceTypeID = '';
+  let publisherNameId = '';
+  let publicationID = '';
+  let dataSetDataTypeID = '';
+  let datasetID = '';
+  let unitsID = '';
+  let reprID = '';
+
+  //to run validation of the json file outline

```

Comment:
We should move the validation functions into a separate file

Reply:
the validations have been moved to validationSchema.ts and is properly now imported to fileUploadService.ts